edkrepo/common/logger.py

```diff
@@ -0,0 +1,143 @@
+import os
+import re
+import git
+import sys
+import time
+import errno
+import string
+import logging
+
+from colorama import init
+from datetime import date
+
+from edkrepo.common.humble import LINE_BREAK, REMOVE_LOG_FAILED, DISK_SPACE_ERROR
+from edkrepo.common.pathfix import expanduser
+from edkrepo.common.edkrepo_exception import EdkrepoLogsRemoveException
+from edkrepo.config.config_factory import GlobalUserConfig
+
+init(autoreset=True)
+
+class CustomFormatter(logging.Formatter):
+    yellow = "\x1b[33;1m"
+    red = "\x1b[31;20m"
+    white = "\x1b[37;10m"
+    FORMATS = {
+        logging.INFO: white,
+        logging.WARNING: yellow,
+        logging.ERROR: red
+    }
+
+    def format(self, record):
+        format_ = "%(message)s\n"
+        if not hasattr(record, 'normal') and not hasattr(record, 'verbose'):
+            record.normal = True
+
+        if hasattr(record, 'safe') and record.safe:
+            safe_str = ''
+            for char in record.msg:
+                if char not in string.printable:
+                    char = '?'
+                safe_str = ''.join((safe_str, str(char)))
+            record.msg = safe_str if safe_str != '' else record.msg
+
+        if hasattr(record,'header') and record.header:
+            format_ = "%(levelname)s: %(message)s\n"
+
+        color = self.FORMATS.get(record.levelno)
+        formatter = logging.Formatter("{}{}".format(color, format_))
+        return formatter.format(record)
+
+class CustomHandler(logging.StreamHandler):
+    def write(self, stream, msg):
+        try:
+            stream.write(msg)
+        except OSError as e:
+            if e.errno == errno.ENOSPC:
+                logger.info(DISK_SPACE_ERROR.format(os.path.join(config.logs_path, file_name)) + ".log")
+
+    def emit(self, record):
+        try:
+            msg = self.format(record)
+            stream = self.stream
+            if hasattr(record, 'verbose') and record.verbose:
+                self.write(stream, msg)
+            if hasattr(record, 'normal') and record.normal:
+                self.write(stream, msg)
+        except:
+            self.handleError(record)
+
+class fileFormatter(logging.Formatter):
+    def format(self, record):

```

Comment:
consider renaming to format_log_file

Reply:
This is overriding the in built format method of logging.Formatter